src/udev/udevadm-wait.c

```diff
@@ -0,0 +1,298 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <getopt.h>
+#include <unistd.h>
+
+#include "sd-event.h"
+
+#include "alloc-util.h"
+#include "chase-symlinks.h"
+#include "errno-util.h"
+#include "fd-util.h"
+#include "inotify-util.h"
+#include "parse-util.h"
+#include "path-util.h"
+#include "static-destruct.h"
+#include "strv.h"
+#include "udev-util.h"
+#include "udevadm.h"
+
+static usec_t arg_timeout_usec = USEC_INFINITY;
+static bool arg_initialized = true;
+static bool arg_removed = false;
+static bool arg_settle = false;
+static char **arg_devices = NULL;
+
+STATIC_DESTRUCTOR_REGISTER(arg_devices, strv_freep);
+
+static int check_device(const char *path) {
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        int r;
+
+        assert(path);
+
+        if (access(path, F_OK) < 0)
+                return arg_removed ? true : -errno;
+
+        if (arg_removed)
+                return false;
+
+        if (!arg_initialized)
+                return true;
+
+        r = sd_device_new_from_path(&dev, path);
```

Comment:
this returns ENODEV if the device doesn't exist. We should return false in that case.